src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,46 @@
+
+package duke.task;
+import duke.exception.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a deadline that extends from the Task class, which consists of a description and a due date.
+ */
+public class Deadline extends Task {
+    private String deadlineDate;
+    private LocalDate deadlineLocalDate;
+    private String formatDeadlineDate;
+
+    public Deadline(String taskName, String deadlineDate) throws DukeException {
+        super(taskName, "D");
+        try {
+            this.deadlineDate = deadlineDate;
+            this.deadlineLocalDate = LocalDate.parse(deadlineDate);
+            this.formatDeadlineDate = this.deadlineLocalDate.format(DateTimeFormatter.ofPattern("dd MMM yyyy"));
+        } catch (Exception e) {
+            throw new DukeException("Please use YYYY-MM-DD format for dates~");
+        }
+    }
+
+    /**
+     * A function to get the due date of the Deadline.
+     * @return a string representing a due date in DD-MMM-YYYY format.
+     */
```

Comment:
The summary of what the function does is great :D, perhaps you can improve it even further by starting your summary with the verb `Gets ...` instead of `a function to ...` . I noticed the same issue at some other places too. You can view more about it here :D [https://se-education.org/guides/conventions/java/intermediate.html#comments](url)

Reply:
Thanks for pointing it out! I definitely need to get used to writing that way :D